src/include/main/db_main.h

```diff
@@ -99,40 +59,642 @@ class DBMain {
+      command_factory_ = std::make_unique<network::PostgresCommandFactory>();
+      provider_ =
+          std::make_unique<network::PostgresProtocolInterpreter::Provider>(common::ManagedPointer(command_factory_));
+      server_ = std::make_unique<network::TerrierServer>(
+          common::ManagedPointer(provider_), common::ManagedPointer(connection_handle_factory_), thread_registry, port);
+    }
+
+    /**
+     * @return ManagedPointer to the component
+     */
+    common::ManagedPointer<network::TerrierServer> GetServer() const { return common::ManagedPointer(server_); }
+
+   private:
+    // Order matters here for destruction order
+    std::unique_ptr<network::ConnectionHandleFactory> connection_handle_factory_;
+    std::unique_ptr<network::PostgresCommandFactory> command_factory_;
+    std::unique_ptr<network::ProtocolInterpreter::Provider> provider_;
+    std::unique_ptr<network::TerrierServer> server_;
+  };
+
+  /**
+   * Creates DBMain objects
+   */
+  class Builder {
+   public:
+    /**
+     * Validate configuration and construct requested DBMain.
+     * @return DBMain that you get to own. YES, YOU! We're just giving away DBMains over here!
+     */
+    std::unique_ptr<DBMain> Build() {
+      auto db_main = std::make_unique<DBMain>();
+
+      std::unique_ptr<settings::SettingsManager> settings_manager = nullptr;
+      if (use_settings_manager_) {
+        TERRIER_ASSERT(!param_map_.empty(), "Settings parameter map was never set.");
+        settings_manager =
+            std::make_unique<settings::SettingsManager>(common::ManagedPointer(db_main), std::move(param_map_));
+
+        record_buffer_segment_size_ =
+            static_cast<uint64_t>(settings_manager->GetInt(settings::Param::record_buffer_segment_size));
+        record_buffer_segment_reuse_ =
+            static_cast<uint64_t>(settings_manager->GetInt(settings::Param::record_buffer_segment_reuse));
+        block_store_size_ = static_cast<uint64_t>(settings_manager->GetInt(settings::Param::block_store_size));
+        block_store_reuse_ = static_cast<uint64_t>(settings_manager->GetInt(settings::Param::block_store_reuse));
+
+        log_file_path_ = settings_manager->GetString(settings::Param::log_file_path);
+        num_log_manager_buffers_ =
+            static_cast<uint64_t>(settings_manager->GetInt64(settings::Param::num_log_manager_buffers));
+        log_serialization_interval_ = settings_manager->GetInt(settings::Param::log_serialization_interval);
+        log_persist_interval_ = settings_manager->GetInt(settings::Param::log_persist_interval);
+        log_persist_threshold_ =
+            static_cast<uint64_t>(settings_manager->GetInt64(settings::Param::log_persist_threshold));
+
+        gc_interval_ = settings_manager->GetInt(settings::Param::gc_interval);
+
+        network_port_ = static_cast<uint16_t>(settings_manager->GetInt(settings::Param::port));
+      }
```

Comment:
Shall we extract this thing into a separate private function? This function is really long at the moment.